src/ndarray/ndarray.cc

```diff
@@ -66,7 +66,8 @@ void TernaryOp(const NDArray &lhs,
       ret.CheckAndAlloc();
       TBlob tmp = ret.data();
       ndarray::Eval<cpu, OP>(lhs.data(), mhs.data(), rhs.data(), &tmp, ctx);
-    }, lhs.ctx(), const_vars, { ret.var() });
+    }, lhs.ctx(), const_vars, { ret.var() },
+    FnProperty::kNormal, 0, false, PROFILER_MESSAGE("TernaryOp"));
```

Comment:
make another micro that stringifies OP so that this print out the op's name instead of a generic TenaryOp.
https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=c%2B%2B%20stringify